commands/system/help.js

```diff
@@ -0,0 +1,326 @@
+const Command = require(`${process.cwd()}/base/Command.js`);
+const { MessageEmbed } = require("discord.js");
+
+const perpage = 10;
+let on = false;
+
+class Help extends Command {
+  constructor(client) {
+    super(client, {
+      name: "help",
+      description: "Get help on a command, command category, or a setting",
+      extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
+      category: "System",
+      usage: "help <category/command/setting>",
+      aliases: ["h", "halp", "commands"],
+    });
+  }
+
+  async run(message, [type], level) {
+    var page = 1;
+    var n = 0;
+    var finalpage;
+    var num;
+
+    const embed = new MessageEmbed()
+      .setTimestamp()
+      .setColor(message.guild.me.roles.highest.color || 5198940)
+      .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL());
+
+    let currentCategory = "";
+    const sorted = this.client.commands.sort((p, c) => (p.help.cat > c.help.cat ? 1 : p.help.name > c.help.name && p.help.cat === c.help.cat ? 1 : -1));
+    if (!type) {
+      const description = `Command category list\n\nUse \`${message.settings.prefix}help <category>\` to find commands for a specific category`;
+      const output = sorted.filter(c => !(level < 10 && c.help.category === "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map((c) => {
```

Comment:
I'd use a `for-of` loop to filter (using `continue;` to skip commands) and doing what you're doing inside the `Array#map` to improve clearlyness. This line is very hard to read and very messy.

Reply:
I used Yukines PR and modified it so it’s paginated, most of the algorithmic stuff is his

Reply:
well thats not mine, its from @AndrewJWin